contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -43,11 +57,133 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
         _erc20Vault = erc20vault_;
         _vault = vault_;
         _marginEngine = IMarginEngine(vault_.marginEngine());
+        _periphery = IPeriphery(vault_.periphery());
+        _vamm = IVAMM(vault_.vamm());
         _currentPosition = vault_.currentPosition();
         _tokens = vault_.vaultTokens();
         _pullExistentials = vault_.pullExistentials();
     }
 
+    event rebalanced(int24 newTickLowerMul, int24 newTickUpperMul);
+
+    function setConstants(int24 logProx, uint256 sigmaWad, uint256 max_possible_lower_bound, int24 tickSpacing) public {
+        _requireAtLeastOperator();
+        _logProximity = logProx;
+        _sigmaWad = sigmaWad;
+        _max_possible_lower_bound = max_possible_lower_bound;
+        _tickSpacing = tickSpacing;
+    }
+
+    function setTickValues(int24 currentTick, int24 tickLower, int24 tickUpper) public {
+        _requireAtLeastOperator();
+        _currentTick = currentTick;
+        _tickLower = tickLower;
+        _tickUpper = tickUpper;
+    }
+
+    /// @notice Get the current tick and position ticks and decide whether to rebalance
+    function rebalanceCheck() public view returns (bool) {
+        
+        // 1. Get current position, lower, and upper ticks
+        // _currentPosition = _vault.currentPosition();
+        // int24 _tickLower = _currentPosition.tickLower;
+        // int24 _tickUpper = _currentPosition.tickUpper;
+
+        // 2. Get current tick
+        // int24 _currentTick = _periphery.getCurrentTick(_periphery.address());
+
+        // 3. Compare current fixed rate to lower and upper bounds
+        if (
+            _tickLower - _logProximity <= _currentTick &&
+            _currentTick <= _tickUpper + _logProximity
+            
+            ) {
+            // 4.1. If current fixed rate is within bounds, return false (don't rebalance)
+            return (false);
+        } else {
+            // 4.2. If current fixed rate is outside bounds, return true (do rebalance)
+            return (true);
+        }
+    }
+
+    /// @notice Get the nearest tick multiple given a tick and tick spacing
+    function nearestTickMultiple(int24 newTick, int24 tickSpacing) internal pure returns (int24) {
```

Comment:
is there a test for this function?

Reply:
The logic seems good. However, `newTick % tickSpacing >= tickSpacing/2` might encounter issues when newTick is negative. It's worth adding UTs to this function and cover cases where newTick is both negative and positive.